### PR TITLE
Support next_n = 4 for SM100 MQA logits

### PR DESCRIPTION
**Pull request by RayWang96**
Use a cluster to process a query instead of only using an SM.

### CODE REVIEW

**Comment by benchislett** [comment]
Any updates on this?

**Comment by RayWang96** [comment]
> Any updates on this?

We have no plan to merge it into main branch
